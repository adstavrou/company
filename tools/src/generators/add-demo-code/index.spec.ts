import { createTreeWithEmptyWorkspace } from '@nx/devkit/testing';
import { Tree } from '@nx/devkit';
import generator from './index';

describe('add-demo base generator', () => {
  let tree: Tree;

  beforeEach(() => {
    tree = createTreeWithEmptyWorkspace();

    // Add basic index.html to simulate demo-code-snippets root
    tree.write(
      'apps/demo-code-snippets/src/index.html',
      `<!DOCTYPE html>
<html>
  <body>
    <ul>
      <!-- DEMO-LIST -->
    </ul>
  </body>
</html>`,
    );
  });

  it('should create basic demo-code-snippets app structure', async () => {
    await generator(tree, { name: 'your-demo' });

    expect(tree.exists('apps/demo-code-snippets/src/your-demo/index.html')).toBe(true);
    expect(tree.exists('apps/demo-code-snippets/src/your-demo/script.ts')).toBe(true);
    expect(tree.exists('apps/demo-code-snippets/src/your-demo/style.css')).toBe(true);

    let indexHtml = tree.read('apps/demo-code-snippets/src/index.html', 'utf-8');
    expect(indexHtml).toContain('<li><a href="./your-demo/index.html">your-demo</a></li>');
    expect(indexHtml).toContain('<!-- DEMO-LIST -->');
  });

  it('should generate expected index.html structure (snapshot)', async () => {
    await generator(tree, { name: 'snapshot-demo' });

    let result = tree.read('apps/demo-code-snippets/src/index.html', 'utf-8');
    expect(result).toMatchSnapshot();
  });
});
