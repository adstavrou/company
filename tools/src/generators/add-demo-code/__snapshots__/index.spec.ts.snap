// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`add-demo base generator should generate expected index.html structure (snapshot) 1`] = `
"<!doctype html>
<html>
  <body>
    <ul>
      <li><a href="./snapshot-demo/index.html">snapshot-demo</a></li>
      <!-- DEMO-LIST -->
    </ul>
  </body>
</html>
"
`;
